Userland/Applications/Browser/BrowserWindow.cpp

```diff
@@ -720,6 +720,10 @@ void BrowserWindow::event(Core::Event& event)
     case GUI::Event::Resize:
         broadcast_window_size(static_cast<GUI::ResizeEvent&>(event).size());
         break;
+    case GUI::Event::KeyDown:
```

Comment:
Hm this is curious. Is there not a GUI::Widget toggle (set_accepts_key_events) or some such that we should be calling?

Or is this how other Serenity GUI apps handle this?

Reply:
Apologies for not responding earlier. From what I gather, it comes down to how the app interacts with the GUI subsystem. I am new to developing here so could be entirely wrong and there is a much better way to do this cleanly.

In majority of the apps (e.g. Flood), they use a `Widget` as shown here: https://github.com/SerenityOS/serenity/blob/ef9bfe3499eccba59d727cb38af8513d43454eb4/Userland/Games/Flood/main.cpp#L86-L87 https://github.com/SerenityOS/serenity/blob/ef9bfe3499eccba59d727cb38af8513d43454eb4/Userland/Games/Flood/MainWidget.h#L13

In some apps (e.g. FlappyBug), the widget is a `Frame` which extends `Widget` and has overridden `keydown_event` to catch the events to deal with or ignore, as show here: https://github.com/SerenityOS/serenity/blob/ef9bfe3499eccba59d727cb38af8513d43454eb4/Userland/Games/FlappyBug/main.cpp#L46 https://github.com/SerenityOS/serenity/blob/ef9bfe3499eccba59d727cb38af8513d43454eb4/Userland/Games/FlappyBug/Game.h#L21

In Browser (and perhaps other apps, though I was fairly comprehensive in https://github.com/SerenityOS/serenity/pull/22591 adding fullscreen), it does not override either `Widget` or `Frame`, but the entire `Window` as show here:
https://github.com/SerenityOS/serenity/blob/ef9bfe3499eccba59d727cb38af8513d43454eb4/Userland/Applications/Browser/main.cpp#L176 https://github.com/SerenityOS/serenity/blob/ef9bfe3499eccba59d727cb38af8513d43454eb4/Userland/Applications/Browser/BrowserWindow.h#L23

`Window` deals with events a bit lower level. It has an `event` which `BrowserWindow` overrides: https://github.com/SerenityOS/serenity/blob/ef9bfe3499eccba59d727cb38af8513d43454eb4/Userland/Applications/Browser/BrowserWindow.h#L63
This is where we are in this patch.
It then calls up the chain, to the `Window::event` which basically splits up the events into their types. A key event like this would go to `handle_key_event` here:
https://github.com/SerenityOS/serenity/blob/ef9bfe3499eccba59d727cb38af8513d43454eb4/Userland/Libraries/LibGUI/Window.cpp#L575
It then dispatches an event to a focussed or main widget (most apps have this main widget):
https://github.com/SerenityOS/serenity/blob/ef9bfe3499eccba59d727cb38af8513d43454eb4/Userland/Libraries/LibGUI/Window.cpp#L585-L588

As there isn't a widget that is going to deal with the event, the event kinda get's swallowed. Adding this change seemed to allow the shortcut keys (F11, F1, alt F4) to work. I'm not sure it's the right place, but that's the rabbit hole I went down ^^^ :)

Reply:
All right. in that case, it sounds like we should be looking at how to make that behavior not get swallowed when overriding Window. but, this change should work for now, thanks for the deep dive!